central/vulnmgmt/service/service_impl.go

```diff
@@ -79,7 +79,16 @@ func (s *serviceImpl) VulnMgmtExportWorkloads(req *v1.VulnMgmtExportWorkloadsReq
 		return errors.Wrap(errox.ServerError, err.Error())
 	}
 
-	return s.deployments.WalkByQuery(ctx, parsedQuery, func(d *storage.Deployment) error {
+	deployments := make([]*storage.Deployment, 0, cacheSize)
```

Comment:
We need another solution for this. This is one of critical functionalities for large scale deployments. It's designed to stream responses from DB directly to gRPC.

If we introduce storing of deployments we are breaking that streaming functionality.

- maybe we can introduce `WalkByTxQuery` where callback will take context
- or change existing signature for callback to include context
- or maybe we can create transaction with context before calling `WalkByQuery` (if that is possible)